File: split/newwhatsit.pas
procedure newwhatsit(s: smallnumber; w: smallnumber);
var
  p: halfword;
begin
  p := getnode(w);
  mem[p].hh.b0 := 8;
  mem[p].hh.b1 := s;
  mem[curlist.tailfield].hh.rh := p;
  curlist.tailfield := p;
end;

File: cpp/newwhatsit.cpp
#include "newwhatsit.h"
#include "getnode.h"

void newwhatsit(smallnumber s, smallnumber w)
{
	auto p = getnode(w);
	type(p) = whatsit_node;
	subtype(p) = s;
	link(curlist.tailfield) = p;
	curlist.tailfield = p;
}
